Fix to address expectation of proper exception

djgraff209 opened the pull request:
This fix changes the WebFluxRequestExecutingMessageHandler
The change specifically changes the contruction of the
WebClientResponseException to use the `create` factory method.

Fixes #3610.

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc).
-->

djgraff209 committed:
Fix to address expectation of proper exception

This fix changes the WebFluxRequestExecutingMessageHandler
The change specifically changes the contruction of the
WebClientResponseException to use the `create` factory method.

This change addresses issue #3610.

pivotal-cla commented:
@djgraff209 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=3611)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-integration&pullRequestId=3611) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

pivotal-cla commented:
@djgraff209 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=3611)!

artembilan reviewed (changes requested):
Please, fix it as we discussed on Gitter: https://gitter.im/spring-projects/spring-integration?at=611069649484630efa2a4c46.

Also consider to cover it somehow with the test. 
See https://github.com/spring-projects/spring-integration/pull/3612 for the place where you could inject the logic for your change.

Thank you!

djgraff209 commented:
Will attempt to provide unit test to cover later today.

djgraff209 committed:
Added changes and unit test to cover updated code

Updated code to use simplified exception construction that is more tolerant
  of larger payloads.
Updated unit tests to check for specific exception types.

djgraff209 committed:
Corrected checkstyle errors

Corrected checkstyle error in imports.
Corrected whitespace between casts.
Corrected trailing whitespace.

artembilan reviewed (approved):
LGTM.
Merging after adding your name to `@author` list.

djgraff209 committed:
Added @author

Added @author entry.

artembilan commented:
Hm.
I meant that you are OK. That's already my responsibility to clean the fix for our standards, but since you have beaten me with that new commit I merge it without my hands 😄 .

The general rule if I request changes, you have to do them. Otherwise that means everything is up to me and you are good.

So, thank you for your patience with that nasty byte buffer issue and contribution.
Looking forward for more!

djgraff209 commented:
Absolutely not a problem. I have been in the corporate world for many years and rigor like I'm trying to follow is meant mainly to reduce the burden on the owners.

Thanks again!

artembilan commented:
... and cherry-picked to `5.4.x` & `5.3.x`
